viewing_party/party.py

```diff
@@ -1,23 +1,206 @@
+
+def watch_movie(user_data, title):
+
+    for movie in user_data["watchlist"]:
+        if title == movie["title"]:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+    return user_data
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+
+    rating_sum = 0 #4.8,8.8
+
+    if len(user_data["watched"]) == 0:
+        return 0
+
+    for movie in user_data["watched"]:
+        rating_sum += (movie["rating"])
+
+    average_rating = rating_sum / len(user_data["watched"])
+#averge rating 0.8
+    return average_rating
+
+        
+def get_most_watched_genre(user_data):
+    genre = {}
+    if user_data["watched"] == []:
+        return None
+
+    for movie in user_data["watched"]:
+        current_movie_genre = movie["genre"]
+        if current_movie_genre not in genre:
+            genre[current_movie_genre] = 1
+        else:
+            genre[current_movie_genre] += 1
+    
+    most_views = 0
+    most_viewed_genre = ""
+    for x in genre:
+        if genre[x] > most_views:
+            most_views = genre[x]
+            most_viewed_genre = x
+    return most_viewed_genre
+
+
+
 
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def get_unique_watched(user_data):
+    friend_movie_common = []
+    unique_movies = []
+    # user_movie_list = []
+
+    for i in range (len(user_data["friends"])):
+
+        for j in range (len(user_data["friends"][i]["watched"])):
+            friend_movie = user_data["friends"][i]["watched"][j]
+            if friend_movie in user_data["watched"]:
+                friend_movie_common.append(friend_movie)
+            # if user_data["watched"] not in friend_movie:
+    for l in range(len(user_data["watched"])):
+        if user_data["watched"][l] not in friend_movie_common:
+            unique_movies.append(user_data["watched"][l])
+
+    return unique_movies
+
+def get_friends_unique_watched(user_data):
+    friend_unique_list = []
+
+    for i in range (len(user_data["friends"])):
+
+        for j in range (len(user_data["friends"][i]["watched"])):
+            friend_movie = user_data["friends"][i]["watched"][j]
+            if friend_movie not in user_data["watched"] and friend_movie not in friend_unique_list:
+                friend_unique_list.append(friend_movie)
+    return friend_unique_list
+
+
+
         
+    
+
+
+
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+# 1. Create a function named `get_available_recs`. This function should...
+# - take one parameter: `user_data`
+
+def get_available_recs(user_data):
+    friends_unique_movies = get_friends_unique_watched(user_data)
+    rec_movies = []
+    for movies in friends_unique_movies:

```

Comment:
`movies` is plural, and there is a convention in programming that a plural name implies that it's a collection holding multiple objects. Since `movies` will hold a dictionary representing a single movie, I'd recommend using singular `movie`.